app/controllers/Challenge.scala

```diff
@@ -266,7 +267,11 @@ final class Challenge(
                       env.challenge.granter(ctx.me, dest, c.perfType.some) flatMap {
                         case Some(denied) =>
                           showChallenge(c, lila.challenge.ChallengeDenied.translated(denied).some)
-                        case None => api.setDestUser(c, dest) inject Redirect(routes.Challenge.show(c.id))
+                        case None =>
+                          if(username == ctx.userId.get)
+                            showChallenge(c, Some(I18nKeys.challenge.cannotChallengeYourself.txt()))
+                          else
+                            api.setDestUser(c, dest) inject Redirect(routes.Challenge.show(c.id))
```

Comment:
I think the check for if the challenge is from yourself should be done with the other checks (e.g. stopping Anon users accepting challenges, and bots etc) and this is done in the `granter`. This will then be a 'denied' challenge and be automatically handled on line 269. So I don't think this function should need to change.

Stylistically if we were to go with this change I would suggest pulling the `if` out to the `case` level, e.g.:
```
case None if username == ctx.userId.get => ...
case None => ... //else path
```

Also, I'm guessing `ctx.userId` is an `Option[User.ID]` and so therefore can be `None`. In which case `.get` is unsafe. I *think* `ctx.userId` has to be defined in the codepath where you are using this, but if you call `.get` on a `None` you get a runtime exception. The safer way of doing this check is to instead wrap the other part into an Option.
e.g. `ctx.userId == Some(username)`
We can discuss `Option` more next week with you if you aren't that familiar with the concept? `Option` was new to me when I came to scala, and its used pretty much everywhere in the codebase, so good to be confident with it.

Finally, good to add a translation, this will still be needed!

Reply:
thank you for the detailed review :+1: 
I created a separate commit so it can be easier to check the changes based upon your review